docs/pages/admin-guides/access-controls/idps/saml-microsoft-entra-external-id.mdx

```diff
@@ -0,0 +1,235 @@
+
+In Azure portal, select **Resource Groups** menu and create a new resource group named `teleport-dev-vm-pool`.
+Make a note of the subscription that you selected for this resource group. We will need this subscription in the next step.
+
+![Entra ID create resource group](../../../../img/access-controls/saml-idp/ms-entra-external-id/entra-create-resource-group.png)
+
+Select `teleport-dev-vm-pool` resource group.
+Then under **Access control (IAM)** menu and click `+ Add` button and select "Add role assignment"
+
+![Entra ID resource group IAM](../../../../img/access-controls/saml-idp/ms-entra-external-id/entra-resource-group-iam.png)
+
+In the role assignment UI, first, select the "Virtual Machine Contributor" role. This role allows assigned users or groups
+to manage VMs in Azure. Next, select the group `teleport-dev-vm` as its member. Finally, create the role assignment.
+
+Your review assignment UI should look like the image below.
+
+![Entra ID resource group role assignment](../../../../img/access-controls/saml-idp/ms-entra-external-id/entra-role-assignment.png)
+
+Click the `Review + assign` button to finish the role assignment.
+
+We've now created a role and resource assignment, where users that are member of `teleport-dev-vm` group can access 
+resource under `teleport-dev-vm-pool` resource group with the privilege granted by the "Virtual Machine Contributor" role.
+
+##  Step 5/5. Link Azure subscription to Microsoft Entra External ID
+
+The Microsoft Entra External ID must be linked with an existing billing subscription. 
+
+This is required so the users can access and billed for Azure resources available under the linked subscription. 
+
+![Microsoft Entra External ID link subscription](../../../../img/access-controls/saml-idp/ms-entra-external-id/entra-link-subscription.png)
+
+In the Azure portal, under **Microsoft Entra External ID** page, link an active subscription to External ID. 
+Make sure to select the same subscription that you selected while creating the resource group in Step 4. 
+
+Once the subscription is linked, user `alice@example.com` will be able to access VM resources available
+under `teleport-dev-vm-pool` resource group.
+
+## Azure CLI access
+
+Users can authenticate Azure CLI with Teleport by providing a tenant ID to `az login` command.
+```bash
+az login --tenant <tenant_id>
+```
+
+The command will open a web browser window that redirects the user to authenticate with Teleport SAML IdP. 
+Upon successful authentication, Microsoft Entra ID responds with credentials for the CLI.
+
+## User access audit logs
+
+User access logs can be found in both Teleport Web UI and in Azure portal.
+
+In Teleport, each user authentication with SAML IdP is logged and the event can be found in the audit log.
+```json
+{
+  "cluster_name": "mytenant.teleport.sh",
+  "code": "TSI000I",
+  "ei": 0,
+  "event": "saml.idp.auth",
+  "service_provider_entity_id": "https://login.microsoftonline.com/<tenant_id>>/",
```

Comment:
We use `<tenant_id>` throughout the page. It might make sense to replace it with a `<Var name="tenant_id">` tag so a user can update it for easier copy/paste.